pinot-plugins/pinot-file-system/pinot-s3/pom.xml

```diff
@@ -39,6 +39,7 @@
     <http.core.version>4.4.13</http.core.version>
     <s3mock.version>2.12.2</s3mock.version>
     <javax.version>3.1.0</javax.version>
+    <woodstox.version>5.3.0</woodstox.version>
```

Comment:
We have the woodstox-core dependency in [pinot-adls ](https://github.com/apache/pinot/blob/master/pinot-plugins/pinot-file-system/pinot-adls/pom.xml#L78) as well. 
Does it make sense to shift this common dependency at the [pinot-file-system ](https://github.com/apache/pinot/blob/master/pinot-plugins/pinot-file-system/pom.xml)level?

Reply:
@abhioncbr @Jackie-Jiang i feel like it'd be better to keep the dependency closer to the plugins that actually need it? for this case, this dep is specifically used by the aws sdk.

```
[2023-10-10 04:21:35.000157] 	at software.amazon.awssdk.services.s3.DefaultS3Client.listObjectsV2(DefaultS3Client.java:5614) ~[pinot-s3-0.13.0-2023-10-09-3ed8e67fe7-SNAPSHOT-shaded.jar:0.13.0-2023-10-09-3ed8e67fe7-SNAPSHOT-3ed8e67fe789f4f16c78c2d9fcad6a6bf27f0a79]
```

Reply:
The dependencies should be added here, but the version is better managed at root level to avoid unnecessary version conflict. I don't know why we have some versions managed here. @KKcorps any specific reason why we have different version here vs root for javax?

Reply:
@Jackie-Jiang @abhioncbr i added a `dependencyManagement` section to make the pinot-file-system plugins use the same version of woodstox, but ran into a conflict in CI:

```
Error:  Dependency convergence error for com.fasterxml.woodstox:woodstox-core:jar:5.3.0 paths to dependency are:
Error:  +-org.apache.pinot:pinot-tools:jar:1.1.0-SNAPSHOT
Error:    +-org.apache.hadoop:hadoop-common:jar:3.2.4:compile
Error:      +-com.fasterxml.woodstox:woodstox-core:jar:5.3.0:compile
Error:  and
Error:  +-org.apache.pinot:pinot-tools:jar:1.1.0-SNAPSHOT
Error:    +-org.apache.pinot:pinot-s3:jar:1.1.0-SNAPSHOT:runtime
Error:      +-com.fasterxml.woodstox:woodstox-core:jar:6.4.0:runtime
```

i think 5.3.0 is the version required by [hadoop-common 3.2.4](https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-common/3.2.4) at the root level of the pinot project. it seems like the option here are to move the `dependencyManagement` version clause to the root level and set as 5.3.0 which downgrades the version used by `pinot-adls`. does that seem reasonable?

Reply:
@dang-stripe Downgrading a dependency is usually risky. Can you check if changing the version in root to `6.4.0` works? Usually libraries are backward compatible.

Reply:
👍 i bumped it to 6.4.0. the release docs say 6.x are all backward compatible w/ 5.x: https://github.com/FasterXML/woodstox/wiki/Woodstox-Release-6.0

> Woodstox 6.0 is the first Java 9 module info compatible version: it is, however, fully API-compatible 5.x and does not require newer JDK version either.

i can also re-test this again on one of our clusters.